Controller/Redirect/Index.php

```diff
@@ -25,10 +25,15 @@ public function execute()
     {
         $satispayPayment = \SatispayGBusiness\Payment::get($this->getRequest()->getParam("payment_id"));
 
+        // Set last transition id as the satispay payment id
+        $order = $this->checkoutSession->getLastRealOrder();
+        $payment = $order->getPayment();
+        $payment->setLastTransId($satispayPayment->id);
```

Comment:
This action is possible only if order and payment are not null.
Move the save of setLastTransId inside the if ($satispayPayment->status == 'ACCEPTED') 
since in the case of not accepted we don't need to save this information.